14_functions_lab/05.py

```diff
@@ -1,10 +1,16 @@
-"""
-Write a function groupby that takes a list
-and a function and returns a dictionary
-keyd by the return value of the function on the list items
-
-For example:
-    groupby(lambda s: s[0], ['foo', 'fi', 'hello', 'hi'])
-    returns: { 'f': ['foo','fi'], 'h': ['hello', 'hi'] }
-"""
-
+"""
+Elad Agam - Functions
+05 - lamda 
+"""
+
+def groupby(function,*args):
+    ret = {}
+    for n in args:
+        if ret.has_key(function(n)):
+            ret[function(n)].append(n)
+        else:
+            ret[function(n)] = [n,]
+    return ret
+    
+
+print groupby(lambda(s): s[0], 'hello', 'hi', 'help', 'bye', 'here')
```

Comment:
This way you run the function(n) twice for every value in args. Could you improve the code to call function(n) only once?